src/client/config.rs

```
@@ -107,9 +116,27 @@ impl Config {
     /// On production setting, the certificate should be added to the local key
     /// storage, using this setting is potentially dangerous.
     ///
-    /// - Defaults to `false`.
+    /// - Defaults to `default`.
     pub fn trust_cert(&mut self) {
-        self.trust_cert = true;
+        if let TrustConfig::CaCertificateLocation(_) = &self.trust {
+            event!(
+                Level::WARN,
+                "trust_ca() was called before. Certificate validation will be disabled.",
+            );
+        }
+        self.trust = TrustConfig::TrustAll;
+    }
+
+    /// If set, the server certificate will be validated against the given CA certificate.
+    /// Mutual exclusive with `trust_cert`
+    ///
+    /// - Defaults to `None`.
+    pub fn trust_cert_ca(&mut self, path: impl ToString) {
+        if let TrustConfig::TrustAll = &self.trust {
+            event!(Level::WARN, "trust_cert() was called before. Ignoring CA.");
```

Same here. Panic instead of ignoring. Remember to mention it in the rustdoc.